dolphinscheduler-listener/dolphinscheduler-listener-plugin/dolphinscheduler-listener-kafka/src/main/java/org/apache/dolphinscheduler/listener/KafkaListener.java

```diff
@@ -0,0 +1,214 @@
+                JSONUtils.toJsonString(taskUpdateEvent));
+    }
+
+    @Override
+    public void onTaskRemoved(TaskRemoveListenerEvent taskRemovedEvent) {
+        sendEvent(taskRemovedEvent.getListenerInstanceParams(), TaskRemoveListenerEvent.class.getSimpleName(),
+                JSONUtils.toJsonString(taskRemovedEvent));
+    }
+
+    @Override
+    public void onTaskStart(TaskStartListenerEvent taskStartEvent) {
+        sendEvent(taskStartEvent.getListenerInstanceParams(), TaskStartListenerEvent.class.getSimpleName(),
+                JSONUtils.toJsonString(taskStartEvent));
+    }
+
+    @Override
+    public void onTaskEnd(TaskEndListenerEvent taskEndEvent) {
+        sendEvent(taskEndEvent.getListenerInstanceParams(), TaskEndListenerEvent.class.getSimpleName(),
+                JSONUtils.toJsonString(taskEndEvent));
+    }
+
+    @Override
+    public void onTaskFail(TaskFailListenerEvent taskErrorEvent) {
+        sendEvent(taskErrorEvent.getListenerInstanceParams(), TaskFailListenerEvent.class.getSimpleName(),
+                JSONUtils.toJsonString(taskErrorEvent));
+    }
+
+    private void sendEvent(Map<String, String> listenerInstanceParams, String key, String value) {
+        String uniqueId = uniqueId(listenerInstanceParams);
+        if (!kafkaProducers.containsKey(uniqueId)) {
+            String kafkaBroker = listenerInstanceParams.get("servers");
+            String username = listenerInstanceParams.get("username");
+            String password = listenerInstanceParams.get("password");
+            Map<String, Object> configurations = new HashMap<>();
+            // TODO: when use username/password, throws exception: Unable to find LoginModule class:
+            // org.apache.kafka.common.security.plain.PlainLoginModule
+            configurations.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBroker);
+            configurations.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+            configurations.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+            if (StringUtils.isNotEmpty(username) && StringUtils.isNotEmpty(password)) {
+                configurations.put("sasl.jaas.config", String.format(
+                        "org.apache.kafka.common.security.plain.PlainLoginModule required username='%s' password='%s';",
+                        username, password));
+                configurations.put("security.protocol", "SASL_PLAINTEXT");
+                configurations.put("sasl.mechanism", "PLAIN");
+            }
+            KafkaProducer<String, String> producer = new KafkaProducer<>(configurations);
+            kafkaProducers.put(uniqueId, producer);
+
+        }
+        KafkaProducer<String, String> producer = kafkaProducers.get(uniqueId);
+        String topic = listenerInstanceParams.get("topic");
+        producer.send(new ProducerRecord<>(topic, key, value), (recordMetadata, e) -> {
```

Comment:
You said you need to `ensures that messages are neither lost nor unordered` but this is asyn send, how can you ensure the message doesn't lost.